web/controllers/v1/bundles_controller.ex

```diff
@@ -41,12 +41,17 @@ defmodule Cog.V1.BundlesController do
 
   def create(conn, %{"bundle" => params}) do
     case install_bundle(params) do
-      {:ok, bundle} ->
+      {:ok, bundle, warnings} ->
+        view = if length(warnings) > 0 do
+          "show_with_warnings.json"
+        else
+          "show.json"
+        end
```

Comment:
Could you do this inside the view instead and conditionally include a `warnings` key in the result?